lib/modbus.js

```diff
@@ -46,11 +49,31 @@ class Modbus extends events.EventEmitter {
       for await (let key of keys) {
         const addressDescriptor = _.get(input, key);
         const address = _.get(addressDescriptor, 'address');
+        // mode function code (default fc4)
+        const fc = addressDescriptor.fc ? addressDescriptor.fc : 4;
+        // modbus read length (default 1)
+        const readlen = addressDescriptor.len ? addressDescriptor.len : 1;
         let value;
         try {
-          value = await this.modbus.readInputRegisters(address, 1);
+          logger.debug(`Polling modbus device ${this.id}: modbusId=${this.modbusId}, fc=${fc}, address=${address}, readlen=${readlen}`);
+          switch (fc) {
```

Comment:
See https://github.com/yaacov/node-modbus-serial/tree/a0a020fa6c0888ffa0b60a845e70b4dfc14d14ec/worker, could leave this switch case to that library.

Reply:
Potentially add "type" to the address descriptor?

Reply:
Hm, I'm also a bit confused about the readlen parameter. It seems that it's always doubled? Readlen of 4 gives me a buffer of 8, readlen 2 a buffer of 4, and readlen 1 a buffer of 2. Maybe describe that somewhere, if that's indeed how it's supposed to work?